arcgis-ios-sdk-samples/Layers/List KML contents/ListKMLContentsSceneViewController.swift

```diff
@@ -0,0 +1,188 @@
+            switch kmlViewpoint.type{
+            case .lookAt:
+                var lookAtPoint = kmlViewpoint.location
+                if kmlViewpoint.altitudeMode != .absolute{
+                    // if the elevation is relative, account for the surface's elevation
+                    let elevation = sceneSurfaceElevation(for: lookAtPoint) ?? 0
+                    lookAtPoint = AGSPoint(x: lookAtPoint.x, y: lookAtPoint.y, z: lookAtPoint.z + elevation, spatialReference: lookAtPoint.spatialReference)
+                }
+                let camera = AGSCamera(lookAt: lookAtPoint,
+                                       distance: kmlViewpoint.range,
+                                       heading: kmlViewpoint.heading,
+                                       pitch: kmlViewpoint.pitch,
+                                       roll: kmlViewpoint.roll)
+                // only the camera parameter is used by the scene
+                return AGSViewpoint(center: kmlViewpoint.location, scale: 1, camera: camera)
+            case .camera:
+                // convert the KML viewpoint to a camera
+                let camera = AGSCamera(location: kmlViewpoint.location,
+                                       heading: kmlViewpoint.heading,
+                                       pitch: kmlViewpoint.pitch,
+                                       roll: kmlViewpoint.roll)
+                // only the camera parameter is used by the scene
+                return AGSViewpoint(center: kmlViewpoint.location, scale: 1, camera: camera)
+            case .unknown:
+                print("Unexpected AGSKMLViewpointType \(kmlViewpoint.type)")
+                return nil
+            }
+        }
+            // the node does not have a predefined viewpoint, so create a viewpoint based on its extent
+        else if let extent = node.extent,
+            // some nodes do not include a geometry, so check that the extent isn't empty
+            !extent.isEmpty {
+            
+            var center = extent.center
+            // take the scene's elevation into account
+            let elevation = sceneSurfaceElevation(for: center) ?? 0
+            
+            // It's possible for `isEmpty` to be false but for width/height to still be zero.
+            if extent.width == 0,
+                extent.height == 0 {
+                
+                center = AGSPoint(x: center.x, y: center.y, z: center.z + elevation, spatialReference: extent.spatialReference)
+                // Defaults based on Google Earth.
+                let camera = AGSCamera(lookAt: center, distance: 1000, heading: 0, pitch: 45, roll: 0)
+                // only the camera parameter is used by the scene
+                return AGSViewpoint(targetExtent: extent, camera: camera)
+                
+            }
+            else {
+                // expand the extent to give some margins when framing the node
+                let bufferRadius = [extent.width, extent.height].max()! / 20
```

Comment:
That is a fascinating way of getting the maximum of two values, but I would recommend you use the global `max(_:_:)` function:
```swift
max(extent.width, extent.height)
```
That way you avoid the force unwrap.

Reply:
Oh right. Global functions exist.